ci/scripts/driver.sh

```diff
@@ -68,24 +69,57 @@ pr_list=$(${GH} pr list --repo "${REPO_URL}" --label "CI-${MACHINE_ID^}-Ready" -
 for pr in ${pr_list}; do
   pr_dir="${GFS_CI_ROOT}/PR/${pr}"
   db_list=$("${ROOT_DIR}/ci/scripts/pr_list_database.py" --add_pr "${pr}" --dbfile "${pr_list_dbfile}")
-  pr_id=0
+  output_ci_single="${GFS_CI_ROOT}/PR/${pr}/output_single.log"
   #############################################################
   # Check if a Ready labeled PR has changed back from once set
-  # and in that case remove all previous jobs in scheduler and
-  # and remove PR from filesystem to start clean
+  # and in that case completely kill the previose driver.sh cron
+  # job and all its decedands as well as  removing all previous
+  # jobs in scheduler and associated files in the PR
   #############################################################
   if [[ "${db_list}" == *"already is in list"* ]]; then
-    pr_id=$("${ROOT_DIR}/ci/scripts/pr_list_database.py" --dbfile "${pr_list_dbfile}" --display "${pr}" | awk '{print $4}') || true
-    pr_id=$((pr_id+1))
-    "${ROOT_DIR}/ci/scripts/pr_list_database.py" --dbfile "${pr_list_dbfile}" --update_pr "${pr}" Open Ready "${pr_id}"
-    for cases in "${pr_dir}/RUNTESTS/"*; do
-      if [[ -z "${cases+x}" ]]; then
-         break
+    # Get the the PID and HOST of the driver.sh cron job
+    # that is stored int he CI database for this PR
+    driver_ID=$("${ROOT_DIR}/ci/scripts/pr_list_database.py" --dbfile "${pr_list_dbfile}" --display "${pr}" | awk '{print $4}') || true
+    driver_PID=$(echo "${driver_ID}" | cut -d":" -f1) || true
+    driver_HOST=$(echo "${driver_ID}" | cut -d":" -f2) || true
+    host_name=$(hostname -s)
+    rm -f "${output_ci_single}"
+    {
+      echo "CI Update on ${MACHINE_ID^} at $(date +'%D %r')" || true
+      echo "================================================="
+      echo "PR:${pr} Reset to ${MACHINE_ID^}-Ready by user and is now restarting CI tests" || true
+    } >> "${output_ci_single}"
+    if [[ "${driver_PID}" -ne 0 ]]; then
+      echo "Driver PID: ${driver_PID} no longer running this build having it killed"
+      if [[ "${driver_HOST}" == "${host_name}"  ]]; then
+        # shellcheck disable=SC2312
+        pstree -A -p "${driver_PID}" | grep -Pow "(?<=\()[0-9]+(?=\))" | xargs kill
+      else
+        # shellcheck disable=SC2312
+        ssh "${driver_HOST}" 'pstree -A -p "${driver_PID}" | grep -Eow "[0-9]+" | xargs kill'
```

Comment:
passwordless ssh in an automated script to kill jobs -- what could possibly go wrong?

Reply:
I know right

Reply:
My ssh'password-less was working fine on Hera and Orion is always on Login Node 1 to it will use the first clause in the IF.  When we go nems account we need to make sure ssh-less-ness is working on Hera.